packages/solo/test/captp-fixture.js

```diff
@@ -44,7 +44,7 @@ export async function makeFixture(PORT, noisy = false) {
       /** @type {() => void} */
       let abortCapTP;
       ws = new WebSocket(
-        `ws://localhost:${PORT}/private/captp?accessToken=${accessToken}`,
+        `ws://127.0.0.1:${PORT}/private/captp?accessToken=${accessToken}`,
```

Comment:
Interesting! I see node now prefers IPv6, but I'm still wondering why that wouldn't work. I guess it doesn't fallback to IPv4 if the server doesn't answer and for some reason we're not listening on IPv6?

Reply:
Indeed - there is no way to fall back to `127.0.0.1` when it's trying to use `::1`. It fails the connection and loops back around trying `::1` again.

Reply:
As for not listening on IPv6 - Generally requesting listen and only providing port enables all IPv4 and IPv6. I did not investigate further, but I would assume that listen here happens explicitly on `127.0.0.1` and that in turn enables only IPv4.

Reply:
Yeah afaik, there is no way to open a single listen socket for both IPv4 and IPv6 restricted to the respective loopback addresses. I believe you can bind to a specific interface without address, but that's very system dependent and I doubt node exposes that. Alternative is to open 2 listen sockets. I suppose another way may be to listen providing `localhost` and hope both the server and client use the same resolution strategies.

Really [Node should implement Happy Eyeballs](https://github.com/nodejs/node/issues/41625), exactly for these kind of cases.